lib/elixir/lib/kernel.ex

```diff
@@ -1860,18 +1865,21 @@ defmodule Kernel do
   like the `all` anonymous function defined above. See `Access.all/0`,
   `Access.key/2` and others as examples.
   """
-  @spec get_and_update_in(Access.t, nonempty_list(term), (term -> {get, term} | :pop)) ::
-                          {get, Access.t} when get: var
+  @spec get_and_update_in(structure :: Access.t, keys :: nonempty_list(any),
+                          (term -> {get_value, update_value :: term} | :pop)) ::
+                          {get_value, Access.t} when get_value: var
   def get_and_update_in(data, keys, fun)
 
-  def get_and_update_in(data, [h], fun) when is_function(h),
+  def get_and_update_in(data, [h], fun) when is_function(h, 3),
```

Comment:
Since we touching all these lines I think it is fine to get rid of one-letter variable?